packages/react/src/transactions/adapters/responders/CreateProfileResponder.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  GetAllProfilesByOwnerAddressDocument,
+  LensApolloClient,
+} from '@lens-protocol/api-bindings';
+import { CreateProfileRequest } from '@lens-protocol/domain/dist/use-cases/profile';
+import { ITransactionResponder } from '@lens-protocol/domain/use-cases/transactions';
+
+export class CreateProfileResponder implements ITransactionResponder<CreateProfileRequest> {
+  constructor(private readonly client: LensApolloClient) {}
+
+  async commit() {
```

Comment:
I couldn't figure out how to check if an existing `GetAllProfilesByOwnerAddress` query was pending with the a partial match on the variables matching `address: activeWallet.address` so for now just invalidating all of the above queries.

Reply:
Ah, this is an interesting approach. Does it work fine?

Reply:
If it were to be a more generic query (e.g. `GetProfile`) I would have some reservation as it might trigger way more queries that it needs to. But here we have a very specific query, so there is a good change that there is at most on active query watching it.

I think we need to be careful of not spreading indiscriminately this approach in other areas as it might cause unnecessary coupling and unwanted network requests and re-rendering, but if it works here seems an elegant solution.

I would test only one thing. Given the `useProfilesOwnedBy` is paginated (infinite scroll pagination implies that the cache contains all pages content up to current). What happen if let's say we are at page 2 when we refresh the query?

You can reduce the page size with `limit` and create as many profile as you want via `useCreateProfile` (see example app UI).

Reply:
interesting, can you double-check if the `refetchQueries` doesn't trigger the `loading` flag to switch back to `true`?

So the code like below doesn't jump back to loading when you trigger profile creation.
```
const { loading, items } = useProfilesOwnedBy();
const { create } = useCreateProfile();

if (loading) {
  return "loading"
}

if (items.length === 0) {
  return // create profile UI
}

const profile = items[0];

return ... // profiel ui

```

Reply:
@cesarenaldi It looks it does not properly include the newly created profile in the paginated result if the query is currently on the 2nd page for example. I think I'll need to rethink this approach

@desfero I checked and it does not trigger the `loading` flag to be switched back to true when it gets refetched